eng/testing/tests.singlefile.targets

```diff
@@ -84,6 +82,28 @@
     </ItemGroup>
   </Target>
 
+  <!--
+    Use init-compiler.sh to locate the compiler toolchain which was resolved for rest of the build.
+    This target is essentially an override hook which is called before `SetupOSSpecificProps` in
+    `Microsoft.NETCore.Native.Unix.targets`. Note that the input is `CppCompilerAndLinker`
+    and the output is `CppLinker`, because from `SetupOSSpecificProps` onwards, we only use `CppLinker`
+    (when alternative compiler, i.e. gcc, is not selected)
+  -->
+  <Target Name="LocateNativeCompiler"
+          Condition="'$(TestNativeAot)' == 'true' and '$(HostOS)' != 'windows'"
+          BeforeTargets="SetupOSSpecificProps">
+      <PropertyGroup>
+        <CppCompilerAndLinker Condition="'$(CppCompilerAndLinker)' == ''">clang</CppCompilerAndLinker>
```

Comment:
looking at the whole PR again, maybe it'd be better to not use this property name at all and go with what we use in the NativeExports.csproj:
```
    <PropertyGroup>
      <NativeCompiler>$(Compiler)</NativeCompiler>
      <NativeCompiler Condition="'$(NativeCompiler)' == ''">clang</NativeCompiler>
    </PropertyGroup>
```

that'd also support overriding via the `$(Compiler)` property.

Reply:
Compiler == CppCompilerAndLinker here, see eng/build.sh

Reply:
omg, so many abstractions 😄

Reply:
:)

I am all for replacing `CppCompilerAndLinker` and `CppLinker` in favor of `NativeCompiler` as property name, but I think some consumers are relying on `CppCompilerAndLinker`.